src/utils/components/EnvironmentEditor/EnvironmentForm.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Button, Form } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+
+import EnvironmentEditor from './components/EnvironmentEditor';
+import EnvironmentFormActions from './components/EnvironmentFormActions';
+import EnvironmentFormSkeleton from './components/EnvironmentFormSkeleton';
+import EnvironmentFormTitle from './components/EnvironmentFormTitle';
+import useEnvironments from './hooks/useEnvironments';
+import useEnvironmentsResources from './hooks/useEnvironmentsResources';
+
+type EnvironmentFormProps = {
+  vm: V1VirtualMachine;
+  onEditChange?: (edited: boolean) => void;
+  updateVM: (updatedVM: V1VirtualMachine) => Promise<V1VirtualMachine | void>;
+};
+
+const EnvironmentForm: React.FC<EnvironmentFormProps> = ({ vm, onEditChange, updateVM }) => {
+  const { t } = useKubevirtTranslation();
+  const ns = vm?.metadata?.namespace;
+
+  const {
+    secrets,
+    configMaps,
+    serviceAccounts,
+    loaded,
+    error: loadError,
+  } = useEnvironmentsResources(ns);
+
+  const {
+    environments,
+    onEnvironmentAdd,
+    onEnvironmentChange,
+    onEnvironmentRemove,
+    onReload,
+    onSave,
+    edited,
+    error: formError,
+    setError,
+  } = useEnvironments(vm, updateVM, onEditChange);
+
+  const environmentNamesSelected = environments.map((env) => env.name);
+
+  const closeError = () => {
```

Comment:
i think closeError is used only once? you can simply send ()=>setError(null) , no need to declare

Reply:
yes. okay